spec/latest/core/index.html

```diff
@@ -744,6 +744,83 @@ <h3>Documenting a Web API</h3>
       may not discover such a construct correctly.</p>
 
+    <p>There are rare situations when some supported operations once announced in the
+      API documentation should be retracted, i.e. to reflect current state of the 
+      application or current user capabilities (or lack of them). It is doable
+      with the <i>availability<i> predicate that can be added to the <i>Operation</i>
+      itself. By default all operations are <i>Available</i> without limits and it
+      is up for the server to change that, and for the client to discover any 
+      discrepancies compared against the API documentation in the runtime. Yet
+      server may want to inform the client that at the very moment an operation is either 
+      <i>Unavailable</i> or <i>Unauthorized</i>. The latter MAY be used to notify 
+      client that the user used to invoke the request that provides operation 
+      availability information is currently not authorized for that very operation 
+      to be invoked. The former on the other hand MAY be used to inform that
+      the operation is currently forbidden and client SHOULD NOT invoke it.
+	  Server MAY also provide more custom reasons for which the operation is not
+	  available but it SHOULD provide at least one of the <i>Hydra</i> built in reasons.</p>
+
+    <p>Usage of the <i>availability</i> is restricted only to the <i>Operation</i>s
+      and MUST NOT be used on <i>SupportedOperation</i>s. The <i>availability</i>
+      predicate also MUST take precedence over a supported operation declaration.
+      The operation that comes with <i>availability</i> predicate SHOULD be identified
+      using any of the below enlisted methods:
+      <ul>
+        <li>Iri</li>
+        <li>exact match of a resource described by <i>possibleStatus</i>, <i>method</i>,
+          <i>expects</i>, <i>returns</i> or combination of those.</li>
+      </ul>
+    </p>
+
+    <pre class="example nohighlight" data-transform="updateExample"
+         title="Retracting operations - declaration of supported operations">
+      <!--
+      {
+        "@context": "http://www.w3.org/ns/hydra/context.jsonld",
+        "@id": "http://api.example.com/doc/#comments",
+        "@type": "Link",
+        "supportedOperation": [
+          {
+            "@type": "Operation",
+            "method": "POST",
+            "expects": "http://api.example.com/doc/#Upload"
+          },
+          {
+            "@type": "Operation",
+            "method": "POST",
+            "expects": "http://api.example.com/doc/#Comment"
+          }
+        ]
+      }
+      -->
+    </pre>
+    <pre class="example nohighlight" data-transform="updateExample"
+         title="Retracting operations - matching">
+      <!--
+      {
+        "@context": "http://www.w3.org/ns/hydra/context.jsonld",
+        "@id": "http://api.example.com/doc/",
+        "@type": "Collection",
+        "operation": [
```

Comment:
Just a quick remark why I don't like how this proposal uses `hydra:operation`

Existing implementations which do not understand the `"availability"` will inevitably interpret as an inline operation.

Thus, the need to separate the declaration of available and unavailable operation IMO

Reply:
I've change the whole approach with a separate predicate which should resolve tis issue